validator/sawtooth_validator/networking/dispatch.py

```diff
@@ -195,11 +202,13 @@ class _ManagerCollection(object):
     def __init__(self, handler_managers):
         self._chain = handler_managers
         self._index = 0
+        self._lock = Lock()
 
     def __next__(self):
-        result = self._chain[self._index]
-        self._index += 1
-        return result
+        with self._lock:
```

Comment:
If this is a shared structure across threads, there's possible a lot wrong with this iteration, that a lock is not going to help:  the index will get advanced by each thread, which may result in them getting different iteration results.  Also, what raises `StopIteration` here?  Looks like right now, it will throw a ValueError instead.